src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -132,10 +146,25 @@ private static TypeSymbol InferReturnType(BoundBlock block, Binder binder, bool
 
             // Async:
 
+            NamedTypeSymbol taskType = null;
+            var delegateReturnType = delegateType?.GetDelegateType()?.DelegateInvokeMethod?.ReturnType as NamedTypeSymbol;
+            if ((object)delegateReturnType != null)
+            {
+                MethodSymbol createBuilderMethod;
+                var delegateTaskBuilderType = delegateReturnType.GetAsyncMethodBuilderType(out createBuilderMethod);
```

Comment:
`delegateReturnType` (rather than `delegateTaskBuilderType`) is assigned to `taskType`. I've removed the `delegateTaskBuilderType` local to avoid confusion.